controllers/bpfman-agent/cl_application_program.go

```diff
@@ -156,56 +150,32 @@ func (r *ClBpfApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 		r.Logger.Info("Reconciling ClusterBpfApplication", "Name", r.currentApp.Name)
 
-		// Get the corresponding BpfApplicationState object, and if it doesn't
-		// exist, instantiate a copy. If bpfAppStateNew is true, then we need to
-		// create a new BpfApplicationState object instead of just updating the
-		// existing one.
-		appState, bpfAppStateNew, err := r.getBpfAppState(ctx, !r.isBeingDeleted())
+		// Get the BpfApplicationState object for this node if it exists.
+		appState, err := r.getBpfAppState(ctx)
 		if err != nil {
 			r.Logger.Error(err, "failed to get BpfApplicationState")
 			return ctrl.Result{}, err
 		}
 
-		if appState == nil && r.isBeingDeleted() {
-			// If the BpfApplicationState doesn't exist and the BpfApplication
-			// is being deleted, we don't need to do anything.  Just continue
-			// with the next BpfApplication.
-			r.Logger.Info("BpfApplicationState doesn't exist and BpfApplication is being deleted",
-				"Name", r.currentApp.Name)
-			continue
+		if appState == nil {
+			if r.isBeingDeleted() {
+				// If the BpfApplicationState doesn't exist and the BpfApplication
+				// is being deleted, we don't need to do anything.  Just continue
+				// with the next BpfApplication.
+				r.Logger.Info("BpfApplicationState doesn't exist and BpfApplication is being deleted",
+					"Name", r.currentApp.Name)
+				continue
+			}
+			// Create a new BpfApplicationState object first, once it's created,
+			// initialize the Status subresource and then update the status.
+			return r.createBpfAppState(ctx)
```

Comment:
nit: comments should say ClBpfApplicationState

Reply:
Done